monai/transforms/traits.py

```diff
@@ -27,23 +29,44 @@ class LazyTrait:
     """
 
     @property
-    def lazy_evaluation(self):
+    def lazy(self):
         """
-        Get whether lazy_evaluation is enabled for this transform instance.
+        Get whether lazy evaluation is enabled for this transform instance.
         Returns:
             True if the transform is operating in a lazy fashion, False if not.
         """
         raise NotImplementedError()
 
-    @lazy_evaluation.setter
-    def lazy_evaluation(self, enabled: bool):
+    @lazy.setter
+    def lazy(self, enabled: bool):
         """
-        Set whether lazy_evaluation is enabled for this transform instance.
+        Set whether lazy evaluation is enabled for this transform instance.
         Args:
             enabled: True if the transform should operate in a lazy fashion, False if not.
         """
         raise NotImplementedError()
 
+    @property
+    def checks_data(self):
```

Comment:
A little more descriptive name would help yes, and commentary explaining what this means given how critical it is. `requires_applied_ops`?

Reply:
Overly-long names are also something of a clarity issue. I think the docstring that goes with it also states quite clearly when it should be set and what it is for.

Reply:
`checks_data` describes what it does. `requires_apply_pending` implies that it will always know what needs to be done in order to handle such classes. My view is that `checks_data` is the better name because the behaviour of the executor in future releases may not require what a name like `requires_apply_pending` implies. The class in question checks the values in the data `checks_data=True`, if it doesn't `checks_data=False`.

Reply:
I would say that a statement of what is needed, not what it does, makes more sense for the name. A transform isn't "checking" data, it requires up-to-date data because it'll use that in a computation. This is a property of a requirement or am I off track here?

Reply:
I agree with @ericspod , "check_data" is too general a name.

Thanks.

Reply:
> I would say that a statement of what is needed, not what it does, makes more sense for the name. A transform isn't "checking" data, it requires up-to-date data because it'll use that in a computation. This is a property of a requirement or am I off track here?

The fact that it needs to evaluate / check the data values, implies that, _as it stands now_, we need to pass it up to date data. I don't want to hold us to that in the future if we find a clever way around it. That's why I am not keen to call it something like "needs_up_to_date_data". It might not in future. If the name can't be "checks_data", then it needs to be something like "evaluates_inputs". That is what it actually does. How the rest of the system handles that may well change over time.